### DISCUSSION TITLE
graphql-ws with SDL and resolvers

### DISCUSSION THREAD

**Discussion by jmc420** [enisdenjo/graphql-ws#458] [Q&A] [answered]
How is a subscription defined in SDL resolved by graphql-ws?

I've put together an example here:

https://github.com/jmc420/graphql_examples/blob/main/graphql-ws/src/server.ts

The schema is defined thus:

```
const gql = `
    type Query {
        sayHello: String!
      }
      
      type Mutation {
        sendMessage(message:String!):String !
      }
      
      type Subscription {
          greeting: String!
      }
    `;

```
The resolvers that work for graph-http are defined thus:

```
const resolvers = {
    sayHello: () => {
        return "Hello"
    },
    sendMessage: ({ message }) => {
        return "You said " + message
    }
}

```
The resolver for graphql-ws requires the word "subscription":

```
const rootValue = {
    subscription: {
        greeting: async function* () {
            for (const hi of ['Hi', 'Bonjour', 'Hola', 'Ciao', 'Zdravo']) {
                yield { greetings: hi };
            }
        }
    }
}

```
When I invoke the subscription via Postman, I get the error:

"Cannot return null for non-nullable field Subscription.greeting"

Why's the greeting returning null?

**Comment by enisdenjo** [comment, marked as answer]
Duplicate of #460.
